src/deque.py

```diff
@@ -0,0 +1,94 @@
+import sys
+
+
+class Node(object):
+    """Creates a Node."""
+    def __init__(self, value, next_up=None, behind=None):
+        self.value = value
+        self.next = next_up
+        self.behind = behind
+
+
+class Deque(object):
+    """Creates a Deque from value or list."""
+    def __init__(self, optional_values=[]):
+        self.head = None
+        self.tail = None
+        self.length = 0
+        self.is_empty = True
+        for x in optional_values:
+            self.add_first(x)
+
+    def add_last(self, value):
```

Comment:
These names should match those in the requirements as you haven't documented otherwise. Downloading this code I would expect an `append` method and I don't get it, but know where does it tell me what I should do.